### PR TITLE
Feature/dsp 42 create initial user classes and models in account service

### PR DESCRIPTION
**Pull request by johnny2565**
Pushed this code for review

### CODE REVIEW

File: src/DwitTech.AccountService.Core/Dtos/UserDto.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        public string Country { get; set; }
+        [Required]
+        public string State { get; set; }
+        [Required]
+        public string City { get; set; }
+        [Required]
+        public string PostalCode { get; set; }
+    }
+
+    //iuserrepository => getuser, createuser, modify and create user values should be manipulated
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Dtos/UserDto.cs]: Remove comments that have no purpose

File: src/DwitTech.AccountService.Core/Dtos/UserCreateDto.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.AccountService.Core.Dtos
+{
+    [AutoMap(typeof(User), ReverseMap =true)]
+    public class UserCreateDto:UserDto
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Dtos/UserCreateDto.cs]: Why do you need to structure your DTOs like this?

File: src/DwitTech.AccountService.Core/Interfaces/IUserService.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.AccountService.Core.Interfaces
+{
+    public interface IUserService
+    {
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Interfaces/IUserService.cs:11]: This is beyond scope of your task.

File: src/DwitTech.AccountService.Core/Services/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        public UserService(UserRepository userRepository, IMapper mapper)
+        {
+            _userRepository = userRepository;
+            _mapper = mapper;
+        }
+
+        public UserReadDto CreateUser(UserCreateDto user)
+        {
+
+            if(user != null)
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/UserService.cs]: This is beyond scope of your task.

File: src/DwitTech.AccountService.Core/Services/UserService.cs [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            var createDb = _userService.CreateUser(userCreateDto);
+
+            return createDb;*/
+
+        }
+
+
+        public void DeleteUser(UserReadDto user)
+        {
+            var readDto = _mapper.Map<User>(user);
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/UserService.cs]: This is beyond scope of your task.

File: src/DwitTech.AccountService.Core/Services/UserService.cs [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+           _userRepository.DeleteUser(readDto);
+        }
+
+        public IEnumerable<UserReadDto> GetAllUsers()
+        {
+            var allUser = _userRepository.GetAll();
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/UserService.cs]: This is beyond scope of your task.

File: src/DwitTech.AccountService.Core/Services/UserService.cs [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+           return _mapper.Map<IEnumerable<UserReadDto>>(allUser);
+        }
+
+
+        public UserReadDto GetUserById(int id)
+        {
+            var foundUser  =  _userRepository.GetUserById(id);
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/UserService.cs]: This is beyond scope of your task.

File: src/DwitTech.AccountService.Core/Services/UserService.cs [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+            return _mapper.Map<UserReadDto>(foundUser);
+            
+  
+        }
+
+
+        public void UpdateUser(UserCreateDto user)
+        {
+            var userUpdateDto = new UserCreateDto
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/UserService.cs]: This is beyond scope of your task.

File: src/DwitTech.AccountService.Data/Entities/User.cs [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        [Required]
+        public string City { get; set; }
+
+        [Required]
+        public string PostalCode { get; set; }
+
+        [Required]
+        public string ZipCode { get; set; }
+
+        [Required]
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Data/Entities/User.cs:54]: Why is Confirm Password in the entity? Meanwhile no password property in the DTO that needs  it

File: src/DwitTech.AccountService.Data/Migrations/20230208165756_InitialMigration.cs [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+                    PhoneNumber = table.Column<string>(type: "character varying(11)", maxLength: 11, nullable: false),
+                    AddressLine1 = table.Column<string>(type: "text", nullable: false),
+                    AddressLine2 = table.Column<string>(type: "text", nullable: false),
+                    Country = table.Column<string>(type: "text", nullable: false),
+                    State = table.Column<string>(type: "text", nullable: false),
+                    City = table.Column<string>(type: "text", nullable: false),
+                    PostalCode = table.Column<string>(type: "text", nullable: false),
+                    ZipCode = table.Column<string>(type: "text", nullable: false),
+                    Password = table.Column<string>(type: "text", nullable: false),
+                    Confirm_Password = table.Column<string>(type: "text", nullable: false),
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Data/Migrations/20230208165756_InitialMigration.cs]: effect of adding confirm password to your entity

File: src/DwitTech.AccountService.Data/repository/IUserRepository.cs [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.AccountService.Data.Repository
+{
+    public interface IUserRepository
+    {
+        IEnumerable<User> GetAll();
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Data/repository/IUserRepository.cs]: This is beyond scope of your task.

File: src/DwitTech.AccountService.Data/repository/UserRepository.cs [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+namespace DwitTech.AccountService.Data.Repository
+{
+    
+
+    
+    public class UserRepository : IUserRepository
+    {
+        private readonly AccountDbContext _accountDbContext;
+
+        List<User> MockUsers = new List<User>(); //Using this as the database
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Data/repository/UserRepository.cs]: I like this

File: src/DwitTech.AccountService.Data/repository/UserRepository.cs [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+        public UserRepository(AccountDbContext accountDbContext)
+        {
+            _accountDbContext = accountDbContext;
+        }
+
+        public User CreateUser(User user)
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Data/repository/UserRepository.cs]: Again the methods here are beyond scope of your task.

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+    public class UserController : BaseController
+    {
+        private readonly UserService _userService;
+        public UserController(UserService userService)
+        {
+            _userService = userService;
+        }
+
+        [HttpGet]
+        public ActionResult GetAllUsers()
```
Comment (dabensymon) [inline src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: This is beyond scope of your task.

**Review by dabensymon — CHANGES_REQUESTED** [review]
Good job but the contents of this PR are mostly beyond scope. We do not require the implementations yet. Just the class structures.

File: src/DwitTech.AccountService.Data/repository/UserRepository.cs [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+namespace DwitTech.AccountService.Data.Repository
+{
+    
+
+    
+    public class UserRepository : IUserRepository
+    {
+        private readonly AccountDbContext _accountDbContext;
+
+        List<User> MockUsers = new List<User>(); //Using this as the database
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Data/repository/UserRepository.cs]: Is this in use? If not, take it out.

File: src/DwitTech.AccountService.Core/Dtos/UserDto.cs [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+        public string AddressLine2 { get; set; }
+        [Required]
+        public string Country { get; set; }
+        [Required]
+        public string State { get; set; }
+        [Required]
+        public string City { get; set; }
+        [Required]
+        public string PostalCode { get; set; }
+    }
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Dtos/UserDto.cs:39]: I believe password is missing here

**Review by dabensymon — CHANGES_REQUESTED** [review]

File: src/DwitTech.AccountService.Core/Dtos/UserDto.cs (hunk 16 above)
Comment (dabensymon) [reply src/DwitTech.AccountService.Core/Dtos/UserDto.cs:39]: This is still pending

**Review by giftedGeorge — APPROVED** [review]

**Review by giftedGeorge — APPROVED** [review]
Looks good to me

**Review by dabensymon — APPROVED** [review]
LGTM... Please check the actions tab and confirm deployment is succesful

**Pull request merged**